evalml/objectives/utils.py

```diff
@@ -214,3 +229,158 @@ def get_core_objectives(problem_type):
         and obj not in get_non_core_objectives()
     ]
     return objectives
+
+
+def get_default_objectives(problem_type, imbalanced=False):
+    """Get the default recommendation score metrics for the given problem type.
+
+    Args:
+        problem_type (str/ProblemType): Type of problem
+        imbalanced (boolean): For multiclass problems, if the classes are imbalanced. Defaults to False
+
+    Returns:
+        Set of string objective names that correspond to ObjectiveBase objectives
+    """
+    problem_type = handle_problem_types(problem_type)
+    objective_list = [
+        obj.name
+        for obj in DEFAULT_RECOMMENDATION_OBJECTIVES
+        if obj.is_defined_for_problem_type(problem_type)
+    ]
+
+    if problem_type == ProblemTypes.MULTICLASS and imbalanced:
+        objective_list.remove(objectives.AUCMicro.name)
```

Comment:
Yeah, I went back and forth on this a lot. I figured this was more concise, but it may also be more unclear. I'm happy to change this around if need be!